bokeh/models/widgets/tables.py

```diff
@@ -239,6 +242,29 @@ class BooleanFormatter(CellFormatter):
     The icon visualizing the check mark.
     """)
 
+class CompositeFormatter(CellFormatter):
+    '''CompositeFormatter applies multiple formatters to a single TableColumn
+
+    '''
+
+    formatters = Sequence(Instance(CellFormatter), default=[], help="""
+        The CompositeFormatter class may be used with any combination of 
+        the installed formatters in the Bokeh library to apply multiple 
+        formatting capabilities (BooleanFormatter, CellFormatter, DateFormatter, 
+        HTMLTemplateFormatter, NumberFormatter, ScientificFormatter, StringFormatter, 
+        to a single TableColumn.
+        
+        Example:
+        example_formatter = CompositeFormatter(formatters=[
+        NumberFormatter(format='0[.]0000'),
+        HTMLTemplateFormatter(template='<a href="https:/www.google.com/search?q=<%='
+                                       ' manufacturer %>+<%= model %>" target="_blank">'
+                                       '<%= value %></a>'),
```

Comment:
Would ask for some formatting changes here, also there is a sphinx directive that will get this formatted in the docs:
```
Example:

    .. code-block:: python
 
    example_formatter = CompositeFormatter(formatters=[
        NumberFormatter(format='0[.]0000'),
        HTMLTemplateFormatter(template='<a href="https:/www.google.com/search?q=<%='
                                       ' manufacturer %>+<%= model %>" target="_blank">'
                                       '<%= value %></a>'),
    ])
```
Also, I think it is important to specify the order in which the formatters will be applied. Is the first formatter closest to the input (applied first) or closest to the output (applied last).

Reply:
I think I am following what you are suggesting here. To do this, would you recommend using something like an ordered list or set to state the order in which the formatters will be applied in the script, or would it be better to write something into the tables.py that would allow the user to set the declare the order they want to prioritize their formatters? Or am I way off-base

I was actually thinking about something similar to this quite a bit last night (I had one of those things are rattling around in my head nights and couldn't sleep). I am probably hugely overthinking this and unintentionally making it more complicated than it is and I think explicitly stating the order would negate this thought, but here is what I had rolling around in the empty space between my ears....

There are combinations of formatters that don't make sense (at least to me) to ever be used together. The most obvious combination that comes to mind for me is If you're using a ScientificFormatter, you wouldn't be using a NumberFormatter or BooleanFormatter. If a user were to try to enter these combinations, is this something that I should consider including as a returning an error for, or are we best off at this point getting a working CompositeFormatter out there, then think about raising errors if necessary?

Reply:
@jensencl The formatters property will preserve an ordered list/sequence already. My comment about stating the order was really about adding a sentence in the docstring that says which one of the formatters in that ordered sequence *we choose* to apply. Some people (esp mathematicians e.g) might expect that the last formatter in the list is the first formatter that the values is passed to. We just need to pick a convention, and document what we chose. 

Regarding the possible incompatible types. That's definitely possibility. Roughly, I think the "input type" for all but the first formatter needs to be "string". There are ways we could attempt to detect and warn about this situation. But I would say we can defer a decision about what, if anything, to do about that. The upside of a modular "building block" architecture like Bokeh's models is that it is very flexible to put all the pieces together in different ways. The downside is that not all combinations of building blocks make sense. It's a judgment call to decide what we can/want to try and detect and warn about.

Reply:
Cleaned up the formatting and added a short sentence to address the order of implementation.